File: Python/create_staircase.py
def create_staircase(integers):
    integers.sort(reverse=True)  # Sort the integers in descending order
    staircase = []  # Initialize the staircase as an empty list
    
    while len(integers) != 0 :  # Continue until the list of integers is empty
        sublist_length = len(staircase) + 1  # Determine the length of the next sublist
        sublist = integers[:sublist_length]  # Extract integers for the next sublist
        
        # Check if the sublist length matches the expected length
        if len(sublist) != sublist_length:
            return False  # Unable to form a staircase
        
        staircase.append(sublist)  # Add the sublist to the staircase
        integers = integers[sublist_length:]  # Remove integers used in the sublist from the original list
    
    return staircase


result = create_staircase( [1,2,3,4,5,6] )
print(result)

result = create_staircase( [1,2,3,4,5,6,7] )
print(result)

File: R/create_staircase.R
create_staircase <- function(integers) {
  integers <- sort(integers, decreasing = TRUE)  # Sort the integers in descending order
  staircase <- list()  # Initialize the staircase as an empty list
  
  while (length(integers) != 0) {  # Continue until the list of integers is empty
    sublist_length <- length(staircase) + 1  # Determine the length of the next sublist
    sublist <- integers[1:sublist_length]  # Extract integers for the next sublist
    
    # Check if the sublist length matches the expected length
    if (length(sublist) != sublist_length) {
      return(FALSE)  # Unable to form a staircase
    }
    
    staircase <- c(staircase, list(sublist))  # Add the sublist to the staircase
    integers <- integers[-c(1:sublist_length)]  # This will remove the first 'sublist_length' elements from the 'integers' list
  }
  
  return(staircase)
}



# Example input
integers <- c(6, 5, 4, 3, 2, 1)

# Call the create_staircase function
staircase <- create_staircase(integers)

# Print the result
print(staircase)